lib/content-services/src/lib/tree/components/tree.component.spec.ts

```diff
@@ -0,0 +1,259 @@
+        spyOn(component.treeService.treeControl, 'isExpanded').and.returnValue(true);
+        fixture.detectChanges();
+        nodeIcons = fixture.debugElement.queryAll(By.css('.adf-icon'));
+        expect(nodeIcons[0].nativeElement.innerText).toContain('chevron_left');
+    });
+
+    it('should emit pagination when nodes are loaded', (done) => {
+        component.treeService.treeNodes = Array.from(treeNodesMockExpanded);
+        component.paginationChanged.subscribe((pagination: PaginationModel) => {
+            expect(pagination.skipCount).toBe(0);
+            expect(pagination.maxItems).toBe(userPreferenceService.paginationSize);
+            done();
+        });
+        component.expandCollapseNode(component.treeService.treeNodes[0]);
+    });
+
+    it('when node has more items to load loadMore node should appear', () => {
+        component.treeService.treeNodes = Array.from(treeNodesMockExpanded);
+        fixture.detectChanges();
+        const loadMoreNode = fixture.nativeElement.querySelector('.adf-tree-load-more-row');
+        expect(loadMoreNode).not.toBeNull();
+    });
+
+    it('should clear the selection and load root nodes on refresh', () => {
+        const selectionSpy = spyOn(component.treeNodesSelection, 'clear');
+        const getNodesSpy = spyOn(component.treeService, 'getSubNodes').and.callThrough();
+        component.refreshTree(0, 25);
+        expect(selectionSpy).toHaveBeenCalled();
+        expect(getNodesSpy).toHaveBeenCalledWith('-root-', 0, 25);
+    });
+
+    it('should call correct server method on collapsing node', () => {
+        component.refreshTree();
+        fixture.detectChanges();
+        const collapseSpy = spyOn(component.treeService, 'collapseNode');
+        spyOn(component.treeService.treeControl, 'isExpanded').and.returnValue(true);
+        getExpandCollapseBtn(component.treeService.treeNodes[0].id).dispatchEvent(new Event('click'));
+        expect(collapseSpy).toHaveBeenCalledWith(component.treeService.treeNodes[0]);
+    });
+
+    it('should call correct server method on expanding node', () => {
+        component.refreshTree();
+        fixture.detectChanges();
+        const collapseSpy = spyOn(component.treeService, 'expandNode');
+        spyOn(component.treeService.treeControl, 'isExpanded').and.returnValue(false);
+        getExpandCollapseBtn(component.treeService.treeNodes[0].id).dispatchEvent(new Event('click'));
+        expect(collapseSpy).toHaveBeenCalledWith(component.treeService.treeNodes[0], treeNodesMockExpanded);
+    });
+
+    it('should load more subnodes and remove load more button when load more button is clicked', () => {
+        component.refreshTree();
+        fixture.detectChanges();
+        spyOn(component.treeService, 'getSubNodes').and.returnValue(of({pagination: {}, entries: Array.from(singleNode)}));
+        const loadMoreBtn = fixture.debugElement.query(By.css('.adf-tree-load-more-button adf-icon')).nativeElement;
```

Comment:
should be targeting the button element, not the icon

Reply:
Same as in the comment above, click is applied to the icon